slackclient/server.py

```diff
@@ -70,6 +84,29 @@ def append_user_agent(self, name, version):
     def rtm_connect(self, reconnect=False, timeout=None, use_rtm_start=True, **kwargs):
         # rtm.start returns user and channel info, rtm.connect does not.
         connect_method = "rtm.start" if use_rtm_start else "rtm.connect"
+
+        # If the `auto_reconnect` param was passed, set the server's `auto_reconnect` attr
+        if kwargs and kwargs["auto_reconnect"] is True:
+            self.auto_reconnect = True
+
+        # If this is an auto reconnect, rate limit reconnect attempts
+        if self.auto_reconnect and reconnect:
+            # Raise a SlackConnectionError after 5 retries within 3 minutes
+            if self.reconnect_attempt == 5:
+                raise SlackConnectionError("RTM connection failed, reached max reconnects.")
+            elif self.reconnect_attempt > 0:
+                # Back off after the the first attempt
+                if (time.time() - self.last_connected_at) < 180:
+                    # Random offset to prevent stampeding reconnects of multiple RTM clients
```

Comment:
Maybe call the variable `backoff_offset_multiplyer` and then you'll be covered and don't need the comment